src/Services/JsonApi.php

```diff
@@ -0,0 +1,31 @@
+<?php
+
+namespace Dxw\Whippet\Services;
+
+// Responsible for:
+// - calling out to a basic API
+// - parsing JSON
+// - converting error codes
+class JsonApi
+{
+    public function __construct(
+        \Dxw\Whippet\Services\BaseApi $base_api
+    ) {
+        $this->baseApi = $base_api;
+    }
+
+    public function get($url)
+    {
+        $response = $this->baseApi->get($url);
+
+        if ($response->isErr()) {
+            return $response;
+        }
+
+        $response_as_array = json_decode($response->unwrap(), true);
+        if (is_null($response_as_array)) {
+            return \Result\Result::err('Received invalid JSON when requesting '.$url);
```

Comment:
OK, I've switched to Guzzle and differentiated connection errors from application errors. Hard to guess what kind of errors we might get: I suggest we wait for real errors to happen and add cases handle them as we go - should be pretty easy in the new setup.